server/swap/swap.go

```diff
@@ -1132,145 +1159,172 @@ func (s *Swapper) makerRedeemStatus(mStatus *swapStatus, tAsset uint32) (makerRe
+		}
+	}
+
+	// That's one less active swap for this order, and a failure. If
+	// this order has no other active swaps, it will be removed from the
+	// order swap tracker by decrementActiveSwapCount since it is
+	// off-book and no new swap negotiations can begin for this order.
+	s.orders.swapFailure(orderAtFault)
+
+	// The other order now has one less active swap too.
+	if s.orders.swapSuccess(otherOrder) {
+		// TODO: We should count this as a successful swap, but should
+		// it only be a completed order with the extra stipulation that
+		// it had already completed another swap?
+		compTime := time.Now().UTC()
+		s.authMgr.RecordCompletedOrder(otherOrder.User(), otherOrder.ID(), compTime)
+		if err := s.storage.SetOrderCompleteTime(otherOrder, encode.UnixMilli(compTime)); err != nil {
+			if db.IsErrGeneralFailure(err) {
+				log.Errorf("fatal error with SetOrderCompleteTime for order %v: %v", otherOrder.UID(), err)
+			} else {
+				log.Warnf("SetOrderCompleteTime for %v: %v", otherOrder.UID(), err)
+			}
+		}
+	}
+
+	// Penalize for failure to act.
+	//
+	// TODO: Arguably, this obviates the RecordCancel above since this
+	// closes the account before the possibility of a cancellation ratio
+	// penalty. I'm keeping it this way for now however since penalties
+	// may become less severe than account closure (e.g. temporary
+	// suspension, cool down, or order throttling), and restored
+	// accounts will still require a record of the revoked order.
+	s.authMgr.Penalize(orderAtFault.User(), account.FailureToAct)
+
+	// Send the revoke_match messages, and solicit acks.
+	s.revoke(match)
+}
+
+func (s *Swapper) checkInactionEventBased() {
 	// If the DB is failing, do not penalize or attempt to start revocations.
 	if err := s.storage.LastErr(); err != nil {
 		log.Errorf("DB in failing state.")
 		return
 	}
 
 	var deletions []*matchTracker
-	oldestAllowed := time.Now().Add(-s.bTimeout).UTC()
+
+	// Do time.Since(event) with the same now time for each match.
+	now := time.Now()
+	tooOld := func(evt time.Time) bool {
+		return now.Sub(evt) >= s.bTimeout
+	}
 
 	checkMatch := func(match *matchTracker) {
 		// Lock entire matchTracker so the following is atomic with respect to
 		// Status.
 		match.mtx.RLock()
 		defer match.mtx.RUnlock()
 
-		if match.makerStatus.swapAsset != assetID && match.takerStatus.swapAsset != assetID {
-			return
-		}
+		log.Tracef("checkInactionEventBased() => checkMatch(%v, %v)",
+			match.ID(), match.Status)
 
 		failMatch := func(makerFault bool) {
-			orderAtFault, otherOrder := match.Taker, order.Order(match.Maker) // an order.Order
-			if makerFault {
-				orderAtFault, otherOrder = match.Maker, match.Taker
+			s.failMatch(match, makerFault)
+			deletions = append(deletions, match)
+		}
+
+		switch match.Status {
+		case order.NewlyMatched:
+			// Maker has not broadcast their swap. They have until match time
+			// plus bTimeout.
+			if tooOld(match.time) {
```

Comment:
`match.time` is set to current time in `readMatches`:
https://github.com/decred/dcrdex/blob/9e720a7650ee7609af4f23718e2a549f51c006ee/server/swap/swap.go#L2528
but the match requests are sent a little later
https://github.com/decred/dcrdex/blob/9e720a7650ee7609af4f23718e2a549f51c006ee/server/swap/swap.go#L2651-L2654

Also, maker may not get the request until `auth.DefaultRequestTimeout + auth.DefaultConnectTimeout` later.

Reply:
The "sent a little later" bit is about as brief as it gets, but as for the request/connect timeouts, that's intentionally deducted from the inaction timeout.  If they aren't connected, that's the point of the inaction check.

Reply:
I also meant to comment that `DefaultRequestTimeout` is perhaps poorly named since it's a **response** timeout that starts when the request is successfully sent.  But I admit there is a peculiar interaction between the response and connect timeouts (a user who is not connected can delay their response by just not being connected when the request is initially generated and queued for send).

Anyway, inaction doesn't kick in until bTimeout after the match request is sent (or queued for send), and I think that's the right way to do this.  If the user is not online and that eats into their action time, that is to be held against them.